### ISSUE TITLE
Creating database replica fails with 404 while the replica is actually created

### ISSUE THREAD

**Issue by Mayeu** [digitalocean/terraform-provider-digitalocean#906]
# Bug Report

## Describe the bug

When trying to create a database replica with PostgreSQL the provider fails with a 404, not adding the replica in the state, while the replica is actually created on DigitalOcean.

### Affected Resource(s)

- digitalocean_database_replica

### Expected Behavior

It should be possible to create PostgreSQL replica from Terraform without error

### Actual Behavior

See "Describe the bug" before, Terraform command output:
```
digitalocean_database_cluster.postgres-example: Still creating... [3m10s elapsed]
digitalocean_database_cluster.postgres-example: Creation complete after 3m19s [id=cef70a33-92e5-407e-b324-e162dc431b90]
digitalocean_database_replica.read-replica: Creating...
digitalocean_database_replica.read-replica: Still creating... [10s elapsed]
digitalocean_database_replica.read-replica: Still creating... [20s elapsed]
╷
│ Error: Error creating DatabaseReplica: Error trying to read DatabaseReplica state: GET https://api.digitalocean.com/v2/databases/cef70a33-92e5-407e-b324-e162dc431b90/replicas/read-replica: 404 (request "252df266-ea7d-4cc4-9ea3-a8bed2db0ab2") not found
│
│   with digitalocean_database_replica.read-replica,
│   on main.tf line 1, in resource "digitalocean_database_replica" "read-replica":
│    1: resource "digitalocean_database_replica" "read-replica" {
│
╵
```

### Steps to Reproduce
1. `terraform apply`

**Terraform Configuration Files**

This is reproducible using the snippet from the [documentation](https://registry.terraform.io/providers/digitalocean/digitalocean/latest/docs/resources/database_replica):
```terraform
resource "digitalocean_database_replica" "read-replica" {
  cluster_id = digitalocean_database_cluster.postgres-example.id
  name       = "read-replica"
  size       = "db-s-1vcpu-1gb"
  region     = "nyc1"
}

resource "digitalocean_database_cluster" "postgres-example" {
  name       = "example-postgres-cluster"
  engine     = "pg"
  version    = "11"
  size       = "db-s-1vcpu-1gb"
  region     = "nyc1"
  node_count = 1
}
```

**Terraform version**
```
Terraform v1.3.5
on darwin_amd64
+ provider registry.terraform.io/digitalocean/digitalocean v2.24.0
```

**Debug Output**
> Please provide a link to a GitHub Gist containing the complete debug output:
https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the
debug output in the issue; just paste a link to the Gist.

This link result in a 404, thus I’m not sure what debug logs exactly you are expecting exactly.

**Panic Output**

No panic in this case.

**Comment by bentranter** [comment]
Hey @Mayeu, thanks for the report, I'm looking into this now.

**Comment by Mayeu** [comment]
@bentranter Thank you 🙏  Don't hesitate to reach out if you need more information 👍

**Comment by Mayeu** [comment]
@bentranter I'm just seeing right now that the PR #907 seems to try to address this issue

**Comment by bentranter** [comment]
Hey @Mayeu, you were right about that PR 😄 the fix is released in [`v2.25.2`](https://github.com/digitalocean/terraform-provider-digitalocean/releases/tag/v2.25.2).

**Comment by Mayeu** [comment]
Hey @bentranter, thank you very much 🙏  I just tried it and I can confirm it solve our issue 👍
